src/main/scala/com/cloudant/clouseau/IndexService.scala

```diff
@@ -228,11 +229,30 @@ class IndexService(ctx: ServiceContext[IndexServiceArgs]) extends Service(ctx) w
         if (IndexWriter.isLocked(dir)) {
           IndexWriter.unlock(dir);
         }
+    }
+    try {
+      val dir = ctx.args.writer.getDirectory
+      removeIndexDir(dir, None)
```

Comment:
why delete the index whenever the process exits?

Reply:
It will not be deleted in every case, only when the directory that holds the index becomes empty.